react-intl/index.d.ts

```diff
@@ -34,14 +34,16 @@ declare namespace ReactIntl {
         formats?: any;
     }
 
-    interface InjectedIntl {
+    interface InjectedIntlProps {
+      intl: {
```

Comment:
If intl isnt null, the typescript compiler will complain while using ur compnent, because prop intl is missing.

The compiler expect:
`<MyComponent intl={*} />`
but injectIntl is doing that job.